### PR TITLE
Bitcore Node Service

### PR DESCRIPTION
**Pull request by braydonf**
Adds an additional module to run BWS as a Bitcore Node Service, such as:

```bash
npm install -g bitcore-node
bitcore-node create mynode
cd mynode
bitcore-node add insight-api
bitcore-node add bitcore-wallet-service
bitcore-node start
```

### CODE REVIEW

**Comment by kleetus** [comment]
I added BWS as a service of Bitcore-node then I merged in the latest from master and everything seems to be working.  LGTM

**Comment by braydonf** [comment]
Great! Updated to the latest master, and have brought the separate processes together.

File: bitcorenode/index.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+ */
+var Service = function(options) {
+  EventEmitter.call(this);
+
+  this.node = options.node;
+  this.https = options.https || this.node.https;
+  this.httpsOptions = options.httpsOptions || this.node.httpsOptions;
+  this.bwsPort = options.bwsPort || Service.BWS_PORT;
+  this.messageBrokerPort = options.messageBrokerPort || Service.MESSAGE_BROKER_PORT;
+  this.lockerPort = options.lockerPort || Service.LOCKER_PORT;
```
Comment (isocolsky) [inline bitcorenode/index.js]: Shouldn't default ports for each service be extracted from `config.js` instead of hardcoded?

Comment (braydonf) [inline bitcorenode/index.js]: That's a good question, we could have the default be based on the `config.js`. 

Configuration options, in this case, can be changed via the `bitcore-node.json` config file, for example:
```json
{
  "servicesConfig": {
    "bitcore-wallet-service": {
      "bwsPort": 3233
    }
  },
}
```

**Pull request merged**